File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/GettingOutRoll.java
package com.adaptionsoft.games.uglytrivia;

public class GettingOutRoll extends Base implements Action {

    private int roll;

    public GettingOutRoll(CurrentPlayer currentPlayer,
            PlayerNames names,
            Places places,
            Purses purses,
            PenaltyBox penaltyBox,
            Categories categories,
            QuestionsByCategory questions,
            boolean[] isGettingOutOfPenaltyBox,
            int roll) {
        super(currentPlayer,
                names,
                places,
                purses,
                penaltyBox,
                categories,
                questions,
                isGettingOutOfPenaltyBox);
        
        this.roll = roll;
    }

    @Override
    public boolean execute() {
        println();
        isGettingOutOfPenaltyBox[0] = true;

        places.add(roll);
        if (places.get(currentPlayer.get()) > 11) places.set(currentPlayer.get(), places.get(currentPlayer.get()) - 12);
        places.printlnPlace();
        
        categories.println();
        questions.ask();

        return false;
    }

    private void println() {
        names.println();
        System.out.println("They have rolled a " + roll);
        System.out.println(names.get() + " is getting out of the penalty box");
    }

}

File: Csharp/Trivia/GettingOutRoll.cs
using System;

namespace UglyTrivia
{
    public class GettingOutRoll : Base, Action
    {
        private int roll;

        public GettingOutRoll(CurrentPlayer currentPlayer, PlayerNames names, Places places, Purses purses, PenaltyBox penaltyBox, Categories categories, QuestionsByCategory questions, bool[] isGettingOutOfPenaltyBox, int roll)
            : base(currentPlayer, names, places, purses, penaltyBox, categories, questions, isGettingOutOfPenaltyBox)
        {
            this.roll = roll;
        }

        public bool Execute()
        {
            Println();
            isGettingOutOfPenaltyBox[0] = true;
            places.Add(roll);
            if (places.Get(currentPlayer.Get()) > 11) places.Set(currentPlayer.Get(), places.Get(currentPlayer.Get()) - 12);
            places.PrintlnPlace();
            categories.Println();
            questions.Ask();
            return false;
        }

        private void Println()
        {
            names.Println();
            Console.WriteLine("They have rolled a " + roll);
            Console.WriteLine(names.Get() + " is getting out of the penalty box");
        }
    }
}
